### PR TITLE
Add: New feature to play audio of a word in dictionary.

### PR DESCRIPTION
**Pull request by volatile-static**
## This branch is working in progress, and we need to adopt more languages before merge.
preview

### introduction
Display a tool button with label “🔊” in popup panel when select a word to translate. 
When click it, it will pronounce the word using the audio from dictionary.
But currently it supports only "Bing" engine.
**We need help to adopt more languages as well as more engines.**
*Welcome to contribute!*

### contribution
- [fork this branch](https://github.com/volatile-static/zotero-pdf-translate/fork) and commit to this pull request
- with ts files in `./src/dict/`, assign the audio URL to `Zotero.ZoteroPDFTranslate._audioSourceURL`.
- optimize UI in `src\view.ts`

### changes
- add global variable `_audioSourceURL` to pass the URL of audio from `requestTranslate` to popup button 'click'
- dynamic `selectionMenu.style.height` according to weather the button is displayed

### discussion
- display to buttons for EN & US?
- a checkbox to determine auto play?
- how to layout the UI of side bar panel and standalone window?

### CODE REVIEW

**Review by windingwind — COMMENTED** [review]
Better give this button a text if possible.
If that makes the work too complex, adding the mp3 link text to the title attribute as a tooltip is also acceptable.

File: src/dict/bingdict.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ async function bingdict(text: string = undefined) {
     },
     (xhr) => {
       let res = xhr.response;
+      Zotero.ZoteroPDFTranslate._audioSourceURL = 
+        res.match(/https?:\/\/\S+\.mp3/);  // match 'http://*.mp3'
```
Comment (windingwind) [inline src/dict/bingdict.ts:15]: Match all audios would be better

**Comment by volatile-static** [comment]
> Better give this button a text if possible. If that makes the work too complex, adding the mp3 link text to the title attribute as a tooltip is also acceptable.

Like this？
<kbd>🔊英</kbd>      <kbd>🔊美</kbd>

**Comment by windingwind** [comment]
> > Better give this button a text if possible. If that makes the work too complex, adding the mp3 link text to the title attribute as a tooltip is also acceptable.
> 
> Like this？ 🔊英 🔊美

That would be nice.

**Comment by volatile-static** [comment]
But we don't know how should it display in other language.

**Comment by zyx335588** [comment]
很好奇，为什么这个PR没有被合并就关闭了，如果只是因为图标问题，标记个US、UK既可以了吧，我想任何想看英文文献的人看到“🔊” 加个US、UK都会明白是什么意思吧

**Pull request closed without merging**